src/index.ts

```diff
@@ -48,6 +49,8 @@ const onAfterSpec = async function (spec: Spec, results: CypressCommandLine.RunR
       console.error(`Failed to report ${spec.name} to Sauce Labs:`, e.message)
     }
   }
+
+  await reporterInstance.reportTestRun(results as RunResult, {jobId: job?.id});
```

Comment:
Maybe better wrap it inside the try-catch block above? Afaik, we'll always need to have a job for reference (according to the current design requirements).